privacyidea/lib/tokens/hotptoken.py

```diff
@@ -255,10 +256,18 @@ def get_init_detail(self, params=None, user=None):
                                                   "img": create_img(oath_url,
                                                                     width=250)
                                                   }
+                                                 
+                    key_bin = binascii.unhexlify(otpkey)
+                    # also strip the padding =, as it will get problems with the google app.
+                    otpkey_human = base64.b32encode(key_bin).strip('=')
+                    response_detail["otpkey"] = {"description": "OTP seed",
```

Comment:
when i will give unhexlify to tokenclass some tests will fail, because e.g. registration token, also goes also through this method, some tokens are not in binary form..., so either we will give this for hotp token or we will need to check and modify each token

Reply:
OK. Happily this is object oriented. So we could inherit the method ``get_init_detail`` in hotp and totp class like this:

~~~~python
def get_init_detail(self, params=None, user=None)
    response_detail = TokenClass.get_init_detail(params, user)
    # somehow add the base32 encoding...
    return response_detail
~~~~

Reply:
we only need to do this in HOTP class, since TOTP is inherited from HOTP. :-)